npe2/io_utils.py

```diff
@@ -65,7 +67,7 @@ def read_get_reader(
 
 def write(
     path: str,
-    layer_data: List[FullLayerData],
+    layer_data: List[Union[FullLayerData, napari.layers.Layer]],
```

Comment:
> What is the difference between FullLayerData and LayerDataTuple? They seem identical to me and I'm curious why both are required/used.

`FullLayerData` is guaranteed to be a three-tuple, `LayerDataTuple` is a union of all valid formats.  defined here

https://github.com/napari/npe2/blob/128cac40ddd6e231840f293b49f2240d34655f04/npe2/types.py#L45-L46

> Could we not import LayerDataTuple instead of declaring a different type in npe2?

that would be the first and only import of napari anywhere in npe2.  I would rather have napari import from npe2 than vice versa.

Reply:
> Would be nice to see a test I think.

I'd rather not have to install the full napari dependency tree just to confirm that `Layer.as_layer_data_tuple()` does indeed turn a layer instance into a three-tuple.  (that should be tested in napari anyway).  So that's why I left it out here.